tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs

```diff
@@ -124,5 +125,34 @@ public void TestUpdateInternal()
             new MyValue("value_new_3").Should().Be(clonedCache[new MyKey("key3")]);
             new MyValue("value_new_2").Should().Be(clonedCache[new MyKey("key2")]);
         }
+
+        [TestMethod]
+        public void TestCacheOverrideIssue2572()
+        {
+            var snapshot = TestBlockchain.GetTestSnapshot();
+            var storages = snapshot.CreateSnapshot();
+
+            storages.Add
+                (
+                new StorageKey() { Key = new byte[] { 0x00, 0x01 }, Id = 0 },
+                new StorageItem() { Value = new byte[] { } }
+                );
+            storages.Add
+                (
+                new StorageKey() { Key = new byte[] { 0x01, 0x01 }, Id = 0 },
+                new StorageItem() { Value = new byte[] { 0x05 } }
+                );
+
+            storages.Commit();
+
+            var item = storages.GetAndChange(new StorageKey() { Key = new byte[] { 0x01, 0x01 }, Id = 0 });
```

Comment:
GetAnChange it's exactly for that, isn't it? It should be 6

Reply:
Snapshot clone yes. But snapshot itself shouldn't. We didn't commit.